bokeh/sphinxext/bokeh_plot.py

```diff
@@ -170,20 +177,22 @@ def run(self):
             (script_tag, js_path, source, docstring, height_hint) = self.process_source(source, path, js_filename)
         except Exception as e:
             raise SphinxError(f"Error generating {js_filename}: \n\n{e}")
-        self.env.bokeh_plot_files.add((js_path, dirname(self.env.docname)))
 
+        self.env.bokeh_plot_files.add((js_path, dirname(self.env.docname)))
         # use the source file name to construct a friendly target_id
         target_id = f"{dashed_docname}.{basename(js_path)}"
         target = [nodes.target("", "", ids=[target_id])]
 
+        sampledata = self.process_sampledata(target, source)
+
         process_docstring = self.options.get("process-docstring", False)
         intro = self.parse(docstring, '<bokeh-content>') if docstring and process_docstring else []
 
         above, below = self.process_code_block(source, docstring)
 
         autoload = [autoload_script(height_hint=height_hint, script_tag=script_tag)]
 
-        return target + intro + above + autoload + below
+        return target + intro + above + autoload + below + sampledata
```

Comment:
I am not quite sure what the returned `xref_nodes` that get added here are, can you elaborate? Offhand I would have just expected `process_sampledata` to populate `env. all_sampledata_xrefs` for the later visitors